dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -610,18 +646,80 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     {
         BOOL foregroundHandled = true;
         THROW_IF_FAILED(InvokeAll(payloadLength, payload, &foregroundHandled));
+        THROW_HR_IF(E_NOT_SET, !foregroundHandled);
+
+        return S_OK;
+    }
+    CATCH_RETURN();
 
-        if (!foregroundHandled)
+    void PushNotificationManager::Run(winrt::IBackgroundTaskInstance const& taskInstance)
+    {
+        auto appProperties = winrt::CoreApplication::Properties();
```

Comment:
Not a big fan of declaring the variable here. It isn't used at all outside of the if-else blocks. I think it's fine to declare it once in each block.

Reply:
This should be moved after the firstNotificationReceived check.